lib/resolveConfig.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const createStylelint = require('./createStylelint');
+const path = require('path');
+
+/**
+ * Resolves the effective configuation for a given file. Resolves to `null` if
+ * no config is found.
+ * @param {string} filePath The path to the file to get the config for.
+ * @param {Pick<
+ *   import('stylelint').LinterOptions,
+ *   | 'cwd'
+ *   | 'config'
+ *   | 'configBasedir'
+ *   | 'configFile'
+ * >} options The options to use when creating the Stylelint instance.
```

Comment:
[suggestion] How about marking the `options` parameter as *optional*?

```suggestion
 * >} [options] - The options to use when creating the Stylelint instance.
```